cirq-google/cirq_google/calibration/workflow.py

```diff
@@ -1022,16 +1049,32 @@ def _make_zeta_chi_gamma_compensation(
                     new_moment_moment_to_calibration == corrections.moment_to_calibration
                 ), f'Inconsistent decompositions with a moment {moment}'
 
-        if other and decompositions:
-            raise IncompatibleMomentError(f'Moment {moment} contains mixed operations')
+        if decompositions:
```

Comment:
this function is getting rather huge and this logic is rather hairy. Clean up ideas

1. refactor line 1003--1050 into a helper function so it's clear what the inputs and outputs for this segment is and you can give it a helpful name. Then you can document the meaning of "decompositions" and "other"
2. otherwise, at least add line comments saying what decompositions and other are
3. This may result in some duplicated code, but can you factor the four possible values of `(bool(decompositions), bool(other))` into four cases or helper functions? the fall-thru logic going on now makes it pretty hard to follow. Once again, helper functions could go a long way to documenting implicitly and explicitly what variables are being set by each block of code.

Reply:
I did slightly different which I think makes more sense: I extracted the entire loop into a separate method. WDYT?

Reply:
Yes, this looks nicer. Although Github's diff is having a hard time with it :)